packages/database/src/utils/__tests__/encryption.test.ts

```diff
@@ -0,0 +1,129 @@
+import { testData } from '../__fixtures__';
+import { createHash, decryptData, encryptData } from '../encryption';
+
+const { hashing, encryption, decryption } = testData;
+
+describe('Hashing', () => {
+  test('Should create hash the valid values', () => {
+    const { valid: validTestCases } = hashing;
+
+    validTestCases.forEach(item => {
+      if (item.expectedHashKey && item.message) {
+        expect(createHash(item.message).toString('hex')).toEqual(
+          item.expectedHashKey,
+        );
+      }
+    });
+  });
+
+  test('Should not create hash the invalid values', () => {
+    const { invalid: invalidTestCases } = hashing;
+
+    invalidTestCases.forEach(item => {
+      expect(() => {
+        createHash(item as any);
+      }).toThrow();
+    });
+  });
+  test('Should create the same hash for the same input', () => {
+    const message = 'sampleMessage';
+    const hash1 = createHash(message).toString('hex');
+    const hash2 = createHash(message).toString('hex');
+
+    expect(hash1).toEqual(hash2);
+  });
+});
+
+describe('Encryption', () => {
+  test('Should encrypt valid data', async () => {
+    const { valid: validTestCases } = encryption;
+
+    for (const testCase of validTestCases) {
+      const { data, key } = testCase;
+
+      for (let i = 0; i < data.length; i += 1) {
+        const _data = data[i];
+        const _key = key[i];
+        if (_data && _key) {
+          if (_data.length === 1 && _key.length === 1) {
+            const encryptedData = encryptData(_data, createHash(_key));
+            expect(encryptedData).toMatch(/^[0-9a-fA-F]+\.[0-9a-fA-F]+$/);
+          } else {
+            const encryptedData1 = await encryptData(_data, createHash(_key));
+            const encryptedData2 = await encryptData(_data, createHash(_key));
+            expect(encryptedData1).not.toEqual(encryptedData2);
+          }
+        }
+      }
+    }
+  });
+
+  test('Should not encrypt invalid data', async () => {
+    const { invalid: invalidTestCases } = encryption;
+
+    for (const testCase of invalidTestCases) {
+      const { data, key } = testCase;
+
+      for (let i = 0; i < data.length; i += 1) {
+        const _data = data[i];
+        const _key = key[i];
+
+        if (_key) {
+          await expect(
+            encryptData(data as any, createHash(_key)),
+          ).rejects.toThrow();
+        } else if (_data) {
+          await expect(encryptData(_data, _key as any)).rejects.toThrow();
+        }
+      }
+    }
+  });
```

Comment:
When having multiple test cases of the same type, group them in `describe` and add individual test cases to `test` block.

```ts
  describe('Should not encrypt invalid data', async () => {
    encryption.invalid.forEach(testCase => {
      test(testCase.name, () => {
         // Write test here
      })
    })
  })
```

Reference: `apps/desktop/src/main/ipc/__tests__/utils.test.ts`